Problem #175: explain of tokens and context.

kegor opened the pull request:
**Problem #175:**
Information about the tokens&contexts is not very clear here https://docs.starcounter.io/v/2.4/topic-guides/blendable-web-apps/view-model-attaching.

**Solution:**
1. I've added an additional explanation about the context nature and main idea, also predefined contexts which are coming with Starcounter were described. I used sentences from the discussion with @warpech. 
2. Few grammatical issues and obvious typos were fixed. 

⚠️ _Note:_
I'm still not 100% sure that I understand this question from the original bug:
> Is there some authority that decides about the tokens and the contexts or is it arbitrary?

— this why I didn't add something new about tokens yet.

Commit:
Problem 175: the description of the contexts in the documentation can be supplemented to reduce questions from users.

Solution: the description has been supplemented, predefined contexts have been described.

Commit:
Problem 175: several grammatical errors and typos were found on the page.

Solution: grammatical errors and typos were fixed.

# Conflicts:
#	docs/topic-guides/blendable-web-apps/view-model-attaching.md

kegor commented:
@warpech, should I use descriptions of the predefined contexts from shared GDrive `Contexts` doc on the view-model-attaching page or they are more likely for communication with the apps team?

warpech commented on docs/topic-guides/blendable-web-apps/view-model-attaching.md:
```diff
@@ -64,7 +64,21 @@ Handlers with empty tokens are called with other handlers with the same empty to
+* `row` — single row of entity data;
+* `icon` — launch icons or smaller icons in other scenarios;
+* `search` — a search result;
+* `app` — a meta context that represents the current app;
```
That's very good but it inspired me to try to make it even clearer:

* `page` — a size factor context. Use it if the view contains full information that the app has about a concept. The view renders in multiple lines. The view is suitable to be attached in full pages about a concept;
* `thumbnail` — a size factor context. Use it if the view contains basic information that the app has about a concept. The view renders in multiple lines. The view is suitable to be attached in side information or teasers about a concept;
* `row` — a size factor context. Use it if the view contains basic information that the app has about a concept. The view renders in a single line. The view is suitable to be attached in lists;
* `icon` — a size factor context. Use it if the view contains the smallest unit of information that the app has about a concept. The view renders in a single element that is a link to a bigger view. The view is suitable to be attached in menu bars;
* `search` — a meta context. Use it if the view is suitable for search results;
* `app` — a meta context. Use it if the view represents the app itself, rather than a concept;

There are some compound contexts, that have a meaning of their own:

* `search`, `row` — a single-line row that is suitable for search results;
* `app`, `icon` — an icon that opens the main page of the app;

warpech commented on docs/topic-guides/blendable-web-apps/view-model-attaching.md:
```diff
@@ -64,7 +64,21 @@ Handlers with empty tokens are called with other handlers with the same empty to
+* `search` — a search result;
+* `app` — a meta context that represents the current app;
+
+As an app author or solution owner, you can come up with his own contexts. That such contexts are unknown to other app developers, so they are less likely to attach view-models from other apps.
```
As an app author or solution owner, you can come up with your own contexts. Keep in mind that such contexts are unknown to other app authors, so they are less likely to attach view-models from other apps.

warpech commented on docs/topic-guides/blendable-web-apps/view-model-attaching.md:
```diff
@@ -64,7 +64,21 @@ Handlers with empty tokens are called with other handlers with the same empty to
+
+As an app author or solution owner, you can come up with his own contexts. That such contexts are unknown to other app developers, so they are less likely to attach view-models from other apps.
+
+You also can use custom contexts in place of the original contexts for fine tuning of the view-model attachment rules. And furthermore contexts can be combined in view-attachment rules, for example `search, row` will be used most commonly for search results and `app, icon` is used to provide app shortcuts for Launchpad.
```
As a solution owner, you can replace the original contexts provided by an app author with custom contexts for fine tuning of the view-model attachment rules.

warpech commented on docs/topic-guides/blendable-web-apps/view-model-attaching.md:
````diff
@@ -123,7 +137,7 @@ and the same but for a given token:
 ```
 
-As you might noticed, a special attachment rule information structure is used here: `BlendingInfo`.  
+As you might be noticed, a special attachment rule information structure is used here: `BlendingInfo`.  
````
As you might have noticed

warpech commented on docs/topic-guides/blendable-web-apps/view-model-attaching.md:
```diff
@@ -64,7 +64,21 @@ Handlers with empty tokens are called with other handlers with the same empty to
+
+No context \(`null` value\) means **match any context**. Otherwise, two handlers are matched if source context contains same elements as destination context. Examples:
 
 * Source context `{ "Readable", "Page" }` is NOT matched with `{ "Writable", "Page" }`.
```
Please change these examples to use `row`, `page`, `search` instead of `Readable`, `Writeable`, `Page`. This will be more relevant to the current usage.

warpech reviewed (changes requested):
Great beginning!

Commit:
Problem 175: text content should be changed by the comments.

Solution: few parts of the page content were updated according to the comments.

kegor commented:
@warpech, changes by the comments were pushed.

warpech reviewed (approved):
@kegor pls merge. This looks good to me as the summary of the current usage of contexts. 

This page will evolve in future, but this is right as of now.

Commit:
Merge remote-tracking branch 'origin/2.4' into 175-explain-of-tokens-and-context

# Conflicts:
#	docs/topic-guides/blendable-web-apps/view-model-attaching.md
